gateway/src/main/java/org/georchestra/gateway/security/GatewaySecurityConfiguration.java

```diff
@@ -61,16 +66,31 @@ public class GatewaySecurityConfiguration {
     @Autowired(required = false)
     ServerLogoutSuccessHandler oidcLogoutSuccessHandler;
 
+    private @Value("${georchestra.gateway.csrfEnabled:false}") boolean csrfEnabled;
+    private @Value("${georchestra.gateway.corsEnabled:false}") boolean corsEnabled;
+
     @Bean
     public SecurityWebFilterChain securityWebFilterChain(ServerHttpSecurity http,
             List<ServerHttpSecurityCustomizer> customizers) throws Exception {
 
         log.info("Initializing security filter chain...");
-        // disable csrf and cors or the websocket connection gets a 403 Forbidden.
-        // Revisit.
-        log.info("CSRF and CORS disabled. Revisit how they interfer with Websockets proxying.");
-        http.csrf().disable().cors().disable();
 
+        if (!csrfEnabled) {
+            log.info("CSRF disabled. Revisit how they interfer with Websockets proxying.");
+            http.csrf().disable();
+        }
+        if (!corsEnabled) {
+            log.info("CORS disabled. Revisit how they interfer with Websockets proxying.");
+            http.cors().disable();
+        } else {
+            CorsConfiguration config = new CorsConfiguration();
+            config.addAllowedOrigin(CorsConfiguration.ALL);
+            config.addAllowedHeader("*");
+            config.addAllowedMethod("*");
+            UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
+            source.registerCorsConfiguration("/**", config);
+            http.cors().configurationSource(source);
+        }
```

Comment:
Another Idea would be to have a list of allowed origin, so you can only allow authorized origins and not all in case you need to pass few URIs.

Reply:
If we go for following the documentation, as @groldan suggested, then maybe there will be no need for it (or the URIs could be added each time the customer asks for it).